org.adempiere.ui.zk/WEB-INF/src/org/adempiere/webui/desktop/DashboardController.java

```diff
@@ -1643,13 +1656,51 @@ private ReportEngine runReport(int AD_Process_ID, int AD_PrintFormat_ID, String
 	 * @throws Exception
 	 */
 	private ReportData generateReport(int AD_Process_ID, int AD_PrintFormat_ID, String parameters, Component component, String contextPath) throws Exception {
+		MProcess process = MProcess.get(Env.getCtx(), AD_Process_ID);
+		File file = null;
+		if(process.getJasperReport() != null) {
+			file = runJasperReport(process, parameters);
+			return new ReportData(new AMedia(process.getName(), "html", "text/html", file, false), -1);
+		}
+			
 		ReportEngine re = runReport(AD_Process_ID, AD_PrintFormat_ID, parameters);
 		if(re == null)
 			return null;
-		File file = FileUtil.createTempFile(re.getName(), ".html");		
+		file = FileUtil.createTempFile(re.getName(), ".html");		
 		re.createHTML(file, false, AEnv.getLanguage(Env.getCtx()), new HTMLExtension(contextPath, "rp", 
 				component.getUuid(), String.valueOf(AD_Process_ID)));
-		return new ReportData(new AMedia(re.getName(), "html", "text/html", file, false), re);
+		return new ReportData(new AMedia(process.getName(), "html", "text/html", file, false), re.getPrintData() != null ? re.getPrintData().getRowCount(false) : 0);
+	}
+
+	private File runJasperReport(MProcess process, String parameters) {
+		MPInstance pInstance = new MPInstance(Env.getCtx(), process.getAD_Process_ID(), 0, 0, null);
+		pInstance.setIsProcessing(true);
+		pInstance.saveEx();
+		try {
+			if(!fillParameter(pInstance, parameters))
+				return null;
+			//
+			ProcessInfo pi = new ProcessInfo (process.getName(), process.getAD_Process_ID(), 0, 0);
+			if(process.getJasperReport() != null) {
```

Comment:
getJasperReport can't be null at this point. should throw runtime exception if it is null.

Reply:
so the logic is if getJasperReport is null, then throw exception, instead run ServerProcessCtl if jasperReport is not null. got it